ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/MainnetPrecompiledContractRegistries.java

```diff
@@ -170,7 +171,16 @@ static void appendPrivacy(
       return;
     }
 
-    if (precompiledContractConfiguration.getPrivacyParameters().isOnchainPrivacyGroupsEnabled()) {
+    if (precompiledContractConfiguration.getPrivacyParameters().isUnrestrictedPrivacyEnabled()) {
+      registry.put(
+          Address.UNRESTRICTED_PRIVACY,
+          accountVersion,
+          new UnrestrictedPrivacyPrecompiledContract(
+              precompiledContractConfiguration.getGasCalculator(),
+              precompiledContractConfiguration.getPrivacyParameters()));
+    } else if (precompiledContractConfiguration
```

Comment:
non-blocking: local var for this?

Reply:
I tried it, it didn't look as nice :smile: